assets/src/date/getTimeSensitiveDisplayDate.js

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { format } from '@wordpress/date';
+
+/**
+ * External dependencies
+ */
+import moment from 'moment-timezone';
+
+/**
+ * Internal dependencies
+ */
+import {
+  DEFAULT_DATE_FORMATTING,
+  getDateFormattedFromWordPressToMoment,
+  getTimeFromNow,
+  isToday,
+  isYesterday,
+} from './';
+
+// Returns hours since date if date passed in matches today
+// Returns yesterday if date passed in is yesterday
+// Returns formatted date + time by default
+/**
+ * @summary                        Formats a date to display relative to time passed since date.
```

Comment:
Left a comment on #3244 about JSDoc. We might need to clarify our usage.

But what I am seeing is that  we probably don't need `@summar` - the full description is suffice.

Inline comments like `// Returns formatted date + time by default` above should be part of the doc block, otherwise they are not really connected to the function.

Also, ideally no empty lines between the doc block & the function

Reply:
Thanks for the details on how the project should ideally be using JSDoc, I'll go ahead and update this. And thanks for spotting the inline, those were just old notes that i forgot to clear out after adding JSDoc - my eyes completely missed they were still there!